### PR TITLE
[IMP] mail, im_livechat: convert livechat to OWL

### PR DESCRIPTION
**Pull request by seb-odoo**
task-2089874

### CODE REVIEW

**Comment by seb-odoo** [comment]
@alexkuhn backend part of livechat, should be done as best as possible without entity

File: addons/im_livechat/models/im_livechat_channel.py [hunk 1]
```diff
@@ -121,6 +122,11 @@ def _get_livechat_mail_channel_vals(self, anonymous_name, operator, user_id=None
             visitor_user = self.env['res.users'].browse(user_id)
             if visitor_user and visitor_user.active:  # valid session user (not public)
                 channel_partner_to_add.append((4, visitor_user.partner_id.id))
+
+        image_path = modules.get_module_resource('mail', 'static', 'src', 'img', 'smiley', 'avatar.jpg')
+        with file_open(image_path, 'rb') as image_file:
+            image_128 = base64.b64encode(image_file.read())
```
Comment (alexkuhn) [inline addons/im_livechat/models/im_livechat_channel.py]: still unsure how to solve this issue, so let's keep older logic instead: use this source when channel of type `livechat`.

File: addons/im_livechat/static/src/owl/components/chat_window_header/chat_window_header.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.component.ChatWindowHeader" t-inherit-mode="extension">
+        <xpath expr="//Icon" position="replace">
+            <t t-if="storeProps.thread.channel_type !== 'livechat'">
+                <t>$0</t>
+            </t>
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/owl/components/chat_window_header/chat_window_header.xml]: Livechat may have an icon, notably when visitor is typing.

File: addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    assert.strictEqual(
+        document.querySelectorAll('.o_Discuss_thread').length,
+        1,
+        "should have thread section inside content"
+    );
+    assert.ok(
+        document.querySelector('.o_Discuss_thread').classList.contains('o_Thread'),
+        "thread section should use thread component"
+    );
+});
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js]: test already in `/mail`, and is unrelated to patches from `/im_livechat`

File: addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            channel_livechat: [{
+                id: 1,
+                channel_type: "livechat",
+                correspondent_name: "Visitor",
+            }],
+        },
+    };
+
+    await this.start({
+        debug: 1,
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js]: no `debug`

File: addons/im_livechat/static/src/owl/components/discuss_sidebar/discuss_sidebar.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+     * @return {mail.store.model.Thread[]}
+     */
+    quickSearchLivechatList() {
+        if (!this.state.quickSearchValue) {
+            return this.storeProps.pinnedLivechatList;
+        }
+        const qsVal = this.state.quickSearchValue.toLowerCase();
+        return this.storeProps.pinnedLivechatList.filter(livechat => {
+            const nameVal = this.storeGetters.threadName(livechat.localId).toLowerCase();
+            return nameVal.indexOf(qsVal) !== -1;
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/owl/components/discuss_sidebar/discuss_sidebar.js]: `.includes()`

File: addons/mail/static/src/owl/components/chat_window_header/chat_window_header.js [hunk 6]
```diff
@@ -20,31 +20,18 @@ class ChatWindowHeader extends Component {
         this.storeProps = useStore((state, props) => {
             const chatWindowLocalId = props.chatWindowLocalId;
             const thread = state.threads[chatWindowLocalId];
-            const threadName = thread
-                ? this.storeGetters.threadName(chatWindowLocalId)
+            // used in threadName getter
```
Comment (alexkuhn) [inline addons/mail/static/src/owl/components/chat_window_header/chat_window_header.js]: then why not use this getter here?

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js (hunk 3 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js]: it was used as reference, removed

File: addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js (hunk 4 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/owl/components/discuss/discuss_tests.js]: ok

File: addons/im_livechat/static/src/owl/components/discuss_sidebar/discuss_sidebar.js (hunk 5 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/owl/components/discuss_sidebar/discuss_sidebar.js]: ok, I changed also the occurrences where I copied it from

File: addons/im_livechat/static/src/owl/components/chat_window_header/chat_window_header.xml (hunk 2 above)
Comment (seb-odoo) [reply addons/im_livechat/static/src/owl/components/chat_window_header/chat_window_header.xml]: I added the icon on the header.

File: addons/mail/static/src/owl/components/chat_window_header/chat_window_header.js (hunk 6 above)
Comment (seb-odoo) [reply addons/mail/static/src/owl/components/chat_window_header/chat_window_header.js]: I misread the old code here, partially reverted

File: addons/im_livechat/models/im_livechat_channel.py (hunk 1 above)
Comment (seb-odoo) [reply addons/im_livechat/models/im_livechat_channel.py]: Done, added the override in thread preview.

After entities it should probably be a getter on thread.

**Comment by seb-odoo** [comment]
All changes done, waiting runbot

**Comment by seb-odoo** [comment]
green except migration

**Review by alexkuhn — APPROVED** [review]

**Pull request merged**